Fixing typos and rewording docs/examples/basic-requirements.md (#9442)
kubernetes/kops: docs/examples/basic-requirements.md

# Common Basic Requirements For Kops-Related Labs. Pre-Flight Check:

Before rushing-in to replicate any of the exercises, please ensure your basic environment is correctly set-up. See [KOPS AWS tutorial](../getting_started/aws.md) for more information.

Basic requirements:

- Configured AWS cli (aws account set-up with proper permissions/roles needed for kops). Depending on your distro, you can set-up directly from packages, or if you want the most updated version, use `pip` (python package manager) to install by running `pip install awscli` command from your local terminal. Your choice!
- Local ssh key ready on `~/.ssh/id_rsa` / `id_rsa.pub`. You can generate it using `ssh-keygen` command if you don't have one already: `ssh-keygen -t rsa -f ~/.ssh/id_rsa -P ""`.
- AWS Region set. 
  - Throughout most of the exercises, we'll deploy our clusters in us-east-1 region (AZs: us-east-1a, us-east-1b, us-east-1c, us-east-1d, us-east-1e and us-east-1f). 

>   - For real HA at the Kubernetes API level, you need 3 masters. 
>   - For even better HA, you'd want to make sure your region of choice has at least as many AZs as the amount of masters in your cluster (i.e. 3 masters <= 3 AZs). 
>     - You can still deploy a multi-master Kubernetes cluster on regions with only 2 AZs or even 1 AZ, but that would result in more than one (and potentially all) masters in the same AZ and thus, if that AZ goes offline, you'll lose two or all your masters. 
>     - You can always check Amazon's regions and AZs availability by following this link: [AWS Global Infrastructure](https://aws.amazon.com/about-aws/global-infrastructure/). 
>   - Remember: The masters are your Kubernetes control plane. If your masters die, you lose control over your Kubernetes cluster.
> - `kubectl` & `kops` installed. This can be done by running the following commands in your terminal (assuming you are using an amd64/x86_64 linux distro):

Reviewer: ```suggestion
  - For real HA at the Kubernetes API level, you need [3 masters](../operations/high_availability.md). 
```

mv kubectl kops  ~/bin
```

> Finally, some of our exercises use the `jq` utility which is available on modern linux distributions. Please ensure to [install](https://stedolan.github.io/jq/download/) it as well.

Reviewer: @MoShitrit could you also delete all the text below this line?

Author: Done 😄 
Thanks for reviewing @hakman !

Reviewer: Np. Thanks for finding these.
@olemarkus does it look better now?

mv kubectl kops  ~/bin
```

> Finally, some of our exercises use the `jq` utility which is available on modern linux distributions. Please ensure to install it as well. Some examples of how to do it:

Reviewer: This should be enough for someone to figure out how to install `jq`. The rest of the text below can be removed.
```suggestion
Finally, some of our exercises use the `jq` utility which is available on modern linux distributions. Please ensure to [install](https://stedolan.github.io/jq/download/) it as well.

```

mv kubectl kops  ~/bin
```

Finally, some of our exercises use the `jq` utility which is available on modern linux distributions. Please ensure to install it as well. Some examples of how to do it:

> **Centos 7:**

Reviewer: ```suggestion
**Amazon Linux, CentOS, RHEL:**
```

yum -y install jq
```

> **Debian 7/8/9, Ubuntu 14.04/16.04:**

Author: Yeah, good point.
I guess we can change that to `**Debian 8/9/10, Ubuntu 16.04/18.04/20/04:**`?

> **Debian 7/8/9, Ubuntu 14.04/16.04:**

Author: Bumped the versions to `**Debian 8/9/10, Ubuntu 16.04/18.04/20.04:**`

> **Debian 7/8/9, Ubuntu 14.04/16.04:**

Author: Actually, @olemarkus I'm just re-reading your comment. Did you mean _completely_ removing the versions and just keeping the distro names, so `Debian/Ubuntu` without specifying versions at all?

> **Debian 9/10, Ubuntu 16.04/18.04/20.04:**

Reviewer: ```suggestion
**Debian, Ubuntu:**
```
